sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -663,6 +667,35 @@ class DataFrame private[sql](
   def orderBy(sortExprs: Column*): DataFrame = sort(sortExprs : _*)
 
   /**
+   * Returns a new [[DataFrame]] partitioned by the given partitioning expressions into
+   * `numPartitions`
+   * `numPartitions` can be < 0 to preserve the current number of partitions.
+   * @group dfops
+   * @since 1.6.0
+   */
+  def distributeBy(partitionExprs: Seq[Column], numPartitions: Int = -1): DataFrame = {
+    PartitionByExpression(partitionExprs.map { _.expr }, logicalPlan, numPartitions)
+  }
```

Comment:
Let's avoid of having default value in the public API (see https://github.com/databricks/scala-style-guide#java-default-param-values). We can overload this.